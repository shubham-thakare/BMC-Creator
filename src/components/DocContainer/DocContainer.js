import React from 'react';
import { DocContainerWrapper } from './styles/DocContainer.style';
import { Icon } from '@blueprintjs/core';

const DocContainer = ({
  children,
  message,
  isEmpty,
  hasEditOption = true,
  allowFullScreen = true,
}) => {
  return (
    <DocContainerWrapper className={isEmpty && 'empty-state'}>
      {hasEditOption && (
        <Icon icon="plus" iconSize={15} className="add-option" />
      )}
      {allowFullScreen && (
        <Icon icon="fullscreen" iconSize={15} className="full-screen-option" />
      )}
      {isEmpty && <span className="message">{message}</span>}
      {children}
    </DocContainerWrapper>
  );
};

export default DocContainer;
