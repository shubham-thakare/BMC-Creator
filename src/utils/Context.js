import { createContext } from 'react';
import htmlToImage from 'html-to-image';
import wait from 'waait';
import cmsData from '../cms';

export const AppContext = createContext({});

const filter = (node) => {
  return node.tagName !== 'i';
};

const exportImage = async () => {
  hideNav();
  await wait(500);
  const node = document.getElementById('canvas-screen');
  closeNav();
  htmlToImage.toSvgDataURL(node, { filter: filter }).then((dataUrl) => {
    var link = document.createElement('a');
    link.download = 'my-image-name.svg';
    link.href = dataUrl;
    link.click();
  });
};

const hideNav = () => {
  document.getElementById('sidebar').style.left = '-220px';
  document.getElementById('canvas-screen').style.paddingLeft = '10px';
};

const openNav = () => {
  document.getElementById('sidebar').style.left = '0';
  document.getElementById('canvas-screen').style.paddingLeft = '230px';
  document.getElementById('open-button').style.display = 'none';
  document.getElementById('toggle-hide').style.display = 'block';
};

const closeNav = () => {
  document.getElementById('sidebar').style.left = '-170px';
  document.getElementById('canvas-screen').style.paddingLeft = '60px';
  document.getElementById('open-button').style.display = 'block';
  document.getElementById('toggle-hide').style.display = 'none';
};

export const contextWithoutState = {
  cmsData,
  exportImage,
  openNav,
  closeNav,
};
