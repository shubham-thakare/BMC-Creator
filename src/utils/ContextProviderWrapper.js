import React, { useReducer } from 'react';
import PropTypes from 'prop-types';
import { ContextProvider } from '.';
import { reducer, appState } from '../reducer';
import { mockState } from '../mock';

export const ContextProviderWrapper = ({ children, useMockState }) => {
  const [state, dispatch] = useReducer(
    reducer,
    useMockState ? mockState : appState,
  );

  return (
    <ContextProvider value={{ state, dispatch }}>{children}</ContextProvider>
  );
};

ContextProviderWrapper.propTypes = {
  useMockState: PropTypes.bool,
};
