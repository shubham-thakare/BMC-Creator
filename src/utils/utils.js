import { createContext } from 'react';
import htmlToImage from 'html-to-image';
import wait from 'waait';
import { UPDATE_STATE_FROM_FILE_DATA } from '../actions';

export const AppContext = createContext({});
export const ContextProvider = AppContext.Provider;
export const ContextConsumer = AppContext.Consumer;

const filter = (node) => {
  return node.tagName !== 'i';
};

export const exportImage = async (fileName) => {
  hideNav();
  await wait(500);
  const node = document.getElementById('canvas-screen');
  closeNav();
  htmlToImage.toSvgDataURL(node, { filter: filter }).then((dataUrl) => {
    const link = document.createElement('a');
    link.download = fileName + '.svg';
    link.href = dataUrl;
    link.click();
  });
};

export const hideNav = () => {
  try {
    document.getElementById('sidebar').style.left = '-220px';
    document.getElementById('canvas-screen').style.paddingLeft = '10px';
  } catch (ex) {}
};

export const openNav = () => {
  try {
    document.getElementById('sidebar').style.left = '0';
    document.getElementById('canvas-screen').style.paddingLeft = '230px';
    document.getElementById('open-button').style.display = 'none';
    document.getElementById('toggle-hide').style.display = 'block';
  } catch (ex) {}
};

export const closeNav = () => {
  try {
    document.getElementById('sidebar').style.left = '-170px';
    document.getElementById('canvas-screen').style.paddingLeft = '60px';
    document.getElementById('open-button').style.display = 'block';
    document.getElementById('toggle-hide').style.display = 'none';
  } catch (ex) {}
};

export const saveFile = (currentState, fileName) => {
  try {
    const pom = document.createElement('a');
    const blob = new Blob([JSON.stringify(currentState)], { type: 'text/xml' });
    pom.setAttribute('href', window.URL.createObjectURL(blob));
    pom.setAttribute('download', fileName + '.bmc');
    pom.dataset.downloadurl = ['text/xml', pom.download, pom.href].join(':');
    pom.draggable = true;
    pom.classList.add('dragout');
    pom.click();
  } catch (ex) {}
};

export const openFile = (dispatch) => {
  try {
    const inputElement = document.createElement('input');
    inputElement.type = 'file';
    inputElement.accept = '.bmc';
    inputElement.addEventListener('change', (event) => {
      const input = event.target;
      const reader = new FileReader();
      reader.onload = () => {
        const data = reader.result;
        dispatch({ action: UPDATE_STATE_FROM_FILE_DATA, payload: data });
      };
      reader.readAsText(input.files[0]);
    });
    inputElement.dispatchEvent(new MouseEvent('click'));
  } catch (ex) {}
};

export const trimString = (string) => {
  return string ? string.toString().trim() : '';
};

export const userVisitedBefore = () => {
  const userVisitedBefore = getCookie('userVisitedBefore');
  setCookie('userVisitedBefore', true, 365);

  return userVisitedBefore;
};

const setCookie = (cname, cvalue, exdays) => {
  const d = new Date();
  d.setTime(d.getTime() + exdays * 24 * 60 * 60 * 1000);
  const expires = 'expires=' + d.toUTCString();
  document.cookie = cname + '=' + cvalue + ';' + expires + ';path=/';
};

const getCookie = (cname) => {
  const name = cname + '=';
  const ca = document.cookie.split(';');
  for (let i = 0; i < ca.length; i++) {
    let c = ca[i];
    while (c.charAt(0) === ' ') {
      c = c.substring(1);
    }
    if (c.indexOf(name) === 0) {
      return c.substring(name.length, c.length);
    }
  }
  return '';
};
