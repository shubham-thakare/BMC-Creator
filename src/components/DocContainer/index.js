import DocContainer from './DocContainer';

export default DocContainer;
