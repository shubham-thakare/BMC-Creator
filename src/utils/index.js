import { AppContext, contextWithoutState } from './Context';

export { AppContext, contextWithoutState };
