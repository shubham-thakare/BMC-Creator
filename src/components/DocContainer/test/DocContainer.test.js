import React from 'react';
import { mount } from 'enzyme';
import DocContainer from '..';
import { ContextProviderWrapper } from '../../../utils';
import { Icon } from '@blueprintjs/core';

let wrapper;

const props = {
  activeKey: 'keyPartners',
  message: 'Test Message',
  isEmpty: true,
  hasEditOption: true,
  hasAddOption: true,
  allowFullScreen: true,
};

describe('DocContainer Component', () => {
  beforeAll(() => {
    wrapper = mount(
      <ContextProviderWrapper useMockState>
        <DocContainer {...props}></DocContainer>
      </ContextProviderWrapper>,
    );
  });

  it('To match snapshot of DocContainer', async (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('Add, Edit, and Fullscreen options exists', async (done) => {
    expect(wrapper.find(Icon)).toHaveLength(3);
    done();
  });

  it('Simulate add operation', async (done) => {
    const add = wrapper.find(Icon).at(1);
    add.simulate('click', { payload: { activeKey: 'keyPartners' } });
    done();
  });

  it('Tips exists', async (done) => {
    expect(wrapper.find('Tips')).toHaveLength(1);
    done();
  });

  it('Notes are being mapped perfectly', async (done) => {
    expect(wrapper.find('Note')).toHaveLength(4);
    done();
  });
});
