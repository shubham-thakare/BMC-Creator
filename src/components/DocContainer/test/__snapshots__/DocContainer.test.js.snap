// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocContainer Component To match snapshot of DocContainer 1`] = `
<ContextProviderWrapper
  useMockState={true}
>
  <DocContainer
    activeKey="keyPartners"
    allowFullScreen={true}
    hasAddOption={true}
    hasEditOption={true}
    isEmpty={true}
    message="Test Message"
  >
    <DocContainerstyle__DocContainerWrapper
      className="empty-state"
    >
      <div
        className="DocContainerstyle__DocContainerWrapper-sc-106t2xe-1 AKFyq empty-state"
      >
        <Blueprint3.Icon
          className="edit-option"
          icon="edit"
          iconSize={15}
        >
          <span
            className="bp3-icon bp3-icon-edit edit-option"
            icon="edit"
          >
            <svg
              data-icon="edit"
              height={15}
              viewBox="0 0 16 16"
              width={15}
            >
              <desc>
                edit
              </desc>
              <path
                d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                fillRule="evenodd"
                key="0"
              />
            </svg>
          </span>
        </Blueprint3.Icon>
        <Blueprint3.Icon
          className="add-option joyride-sixth-step"
          icon="plus"
          iconSize={15}
          onClick={[Function]}
        >
          <span
            className="bp3-icon bp3-icon-plus add-option joyride-sixth-step"
            icon="plus"
            onClick={[Function]}
          >
            <svg
              data-icon="plus"
              height={15}
              viewBox="0 0 16 16"
              width={15}
            >
              <desc>
                plus
              </desc>
              <path
                d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                fillRule="evenodd"
                key="0"
              />
            </svg>
          </span>
        </Blueprint3.Icon>
        <Blueprint3.Icon
          className="full-screen-option"
          icon="fullscreen"
          iconSize={15}
        >
          <span
            className="bp3-icon bp3-icon-fullscreen full-screen-option"
            icon="fullscreen"
          >
            <svg
              data-icon="fullscreen"
              height={15}
              viewBox="0 0 16 16"
              width={15}
            >
              <desc>
                fullscreen
              </desc>
              <path
                d="M3.41 2H5c.55 0 1-.45 1-1s-.45-1-1-1H1C.45 0 0 .45 0 1v4c0 .55.45 1 1 1s1-.45 1-1V3.41L5.29 6.7c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L3.41 2zM6 9c-.28 0-.53.11-.71.29L2 12.59V11c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1h4c.55 0 1-.45 1-1s-.45-1-1-1H3.41l3.29-3.29c.19-.18.3-.43.3-.71 0-.55-.45-1-1-1zm9 1c-.55 0-1 .45-1 1v1.59L10.71 9.3A.965.965 0 0010 9a1.003 1.003 0 00-.71 1.71l3.3 3.29H11c-.55 0-1 .45-1 1s.45 1 1 1h4c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1zm0-10h-4c-.55 0-1 .45-1 1s.45 1 1 1h1.59l-3.3 3.29a1.003 1.003 0 001.42 1.42L14 3.41V5c0 .55.45 1 1 1s1-.45 1-1V1c0-.55-.45-1-1-1z"
                fillRule="evenodd"
                key="0"
              />
            </svg>
          </span>
        </Blueprint3.Icon>
        <Tips>
          <DocContainerstyle__TipsWrapper>
            <div
              className="DocContainerstyle__TipsWrapper-sc-106t2xe-0 fKaaIx"
            >
              <span
                className="message text-bold text-italic"
              >
                Useful Tips
              </span>
              <ul>
                <li>
                  <span
                    className="message"
                  >
                    Update Canvas Title, Version, and Effective Date from the header
                  </span>
                </li>
                <li>
                  <span
                    className="message"
                  >
                    Click on section title to see hints
                  </span>
                </li>
                <li>
                  <span
                    className="message"
                  >
                    Click on
                     
                    <code>
                      +
                    </code>
                     
                    button placed at top right to add new notes
                  </span>
                </li>
                <li>
                  <span
                    className="message"
                  >
                    Click on note to see menu
                  </span>
                </li>
                <li>
                  <span
                    className="message"
                  >
                    Open side navigation bar to
                  </span>
                  <ul>
                    <li>
                      Save File
                    </li>
                    <li>
                      Open Existing File
                    </li>
                    <li>
                      Start Presentation Mode
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
          </DocContainerstyle__TipsWrapper>
        </Tips>
        <Note
          activeKey="keyPartners"
          item={
            Object {
              "background": "red",
              "color": "white",
              "description": "Test description",
              "key": 1587385449790,
              "title": "Test note 1",
            }
          }
          key="1587385449790"
        >
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            content={
              <NoteMenu
                activeKey="keyPartners"
                noteKey={1587385449790}
              />
            }
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="right-bottom"
            targetClassName="full-width"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <Manager>
              <span
                className="bp3-popover-wrapper"
              >
                <Reference
                  innerRef={[Function]}
                >
                  <InnerReference
                    innerRef={[Function]}
                    setReferenceNode={[Function]}
                  >
                    <Blueprint3.ResizeSensor
                      onResize={[Function]}
                    >
                      <span
                        className="bp3-popover-target full-width"
                        onClick={[Function]}
                      >
                        <Notestyle__NoteWrapper
                          bgColor="red"
                          className=""
                          color="white"
                          key=".0"
                        >
                          <div
                            className="Notestyle__NoteWrapper-sc-2z8c2g-0 hYtmWh"
                            color="white"
                          >
                            <Notestyle__NoteTitle>
                              <span
                                className="Notestyle__NoteTitle-sc-2z8c2g-1 gNJdcF"
                              >
                                Test note 1
                              </span>
                            </Notestyle__NoteTitle>
                            <Notestyle__NoteDescription>
                              <span
                                className="Notestyle__NoteDescription-sc-2z8c2g-2 ilDHHe"
                              >
                                Test description
                              </span>
                            </Notestyle__NoteDescription>
                          </div>
                        </Notestyle__NoteWrapper>
                      </span>
                    </Blueprint3.ResizeSensor>
                  </InnerReference>
                </Reference>
                <Blueprint3.Overlay
                  autoFocus={true}
                  backdropClassName="bp3-popover-backdrop"
                  backdropProps={Object {}}
                  canEscapeKeyClose={true}
                  canOutsideClickClose={true}
                  enforceFocus={true}
                  hasBackdrop={false}
                  isOpen={false}
                  lazy={true}
                  onClose={[Function]}
                  transitionDuration={300}
                  transitionName="bp3-popover"
                  usePortal={true}
                />
              </span>
            </Manager>
          </Blueprint3.Popover>
        </Note>
        <Note
          activeKey="keyPartners"
          item={
            Object {
              "background": "red",
              "color": "white",
              "description": "Test description",
              "key": 1587385449791,
              "title": "Test note 2",
            }
          }
          key="1587385449791"
        >
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            content={
              <NoteMenu
                activeKey="keyPartners"
                noteKey={1587385449791}
              />
            }
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="right-bottom"
            targetClassName="full-width"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <Manager>
              <span
                className="bp3-popover-wrapper"
              >
                <Reference
                  innerRef={[Function]}
                >
                  <InnerReference
                    innerRef={[Function]}
                    setReferenceNode={[Function]}
                  >
                    <Blueprint3.ResizeSensor
                      onResize={[Function]}
                    >
                      <span
                        className="bp3-popover-target full-width"
                        onClick={[Function]}
                      >
                        <Notestyle__NoteWrapper
                          bgColor="red"
                          className=""
                          color="white"
                          key=".0"
                        >
                          <div
                            className="Notestyle__NoteWrapper-sc-2z8c2g-0 hYtmWh"
                            color="white"
                          >
                            <Notestyle__NoteTitle>
                              <span
                                className="Notestyle__NoteTitle-sc-2z8c2g-1 gNJdcF"
                              >
                                Test note 2
                              </span>
                            </Notestyle__NoteTitle>
                            <Notestyle__NoteDescription>
                              <span
                                className="Notestyle__NoteDescription-sc-2z8c2g-2 ilDHHe"
                              >
                                Test description
                              </span>
                            </Notestyle__NoteDescription>
                          </div>
                        </Notestyle__NoteWrapper>
                      </span>
                    </Blueprint3.ResizeSensor>
                  </InnerReference>
                </Reference>
                <Blueprint3.Overlay
                  autoFocus={true}
                  backdropClassName="bp3-popover-backdrop"
                  backdropProps={Object {}}
                  canEscapeKeyClose={true}
                  canOutsideClickClose={true}
                  enforceFocus={true}
                  hasBackdrop={false}
                  isOpen={false}
                  lazy={true}
                  onClose={[Function]}
                  transitionDuration={300}
                  transitionName="bp3-popover"
                  usePortal={true}
                />
              </span>
            </Manager>
          </Blueprint3.Popover>
        </Note>
        <Note
          activeKey="keyPartners"
          item={
            Object {
              "background": undefined,
              "color": "white",
              "description": "Test description",
              "key": 1587385449792,
              "title": "",
            }
          }
          key="1587385449792"
        >
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            content={
              <NoteMenu
                activeKey="keyPartners"
                noteKey={1587385449792}
              />
            }
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="right-bottom"
            targetClassName="full-width"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <Manager>
              <span
                className="bp3-popover-wrapper"
              >
                <Reference
                  innerRef={[Function]}
                >
                  <InnerReference
                    innerRef={[Function]}
                    setReferenceNode={[Function]}
                  >
                    <Blueprint3.ResizeSensor
                      onResize={[Function]}
                    >
                      <span
                        className="bp3-popover-target full-width"
                        onClick={[Function]}
                      >
                        <Notestyle__NoteWrapper
                          className=""
                          color="white"
                          key=".0"
                        >
                          <div
                            className="Notestyle__NoteWrapper-sc-2z8c2g-0 hiIQbo"
                            color="white"
                          >
                            <Notestyle__NoteDescription>
                              <span
                                className="Notestyle__NoteDescription-sc-2z8c2g-2 ilDHHe"
                              >
                                Test description
                              </span>
                            </Notestyle__NoteDescription>
                          </div>
                        </Notestyle__NoteWrapper>
                      </span>
                    </Blueprint3.ResizeSensor>
                  </InnerReference>
                </Reference>
                <Blueprint3.Overlay
                  autoFocus={true}
                  backdropClassName="bp3-popover-backdrop"
                  backdropProps={Object {}}
                  canEscapeKeyClose={true}
                  canOutsideClickClose={true}
                  enforceFocus={true}
                  hasBackdrop={false}
                  isOpen={false}
                  lazy={true}
                  onClose={[Function]}
                  transitionDuration={300}
                  transitionName="bp3-popover"
                  usePortal={true}
                />
              </span>
            </Manager>
          </Blueprint3.Popover>
        </Note>
        <Note
          activeKey="keyPartners"
          item={
            Object {
              "background": "red",
              "color": "white",
              "description": "",
              "key": 1587385449793,
              "title": "Test note 2",
            }
          }
          key="1587385449793"
        >
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            content={
              <NoteMenu
                activeKey="keyPartners"
                noteKey={1587385449793}
              />
            }
            defaultIsOpen={false}
            disabled={false}
            fill={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="right-bottom"
            targetClassName="full-width"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <Manager>
              <span
                className="bp3-popover-wrapper"
              >
                <Reference
                  innerRef={[Function]}
                >
                  <InnerReference
                    innerRef={[Function]}
                    setReferenceNode={[Function]}
                  >
                    <Blueprint3.ResizeSensor
                      onResize={[Function]}
                    >
                      <span
                        className="bp3-popover-target full-width"
                        onClick={[Function]}
                      >
                        <Notestyle__NoteWrapper
                          bgColor="red"
                          className=""
                          color="white"
                          key=".0"
                        >
                          <div
                            className="Notestyle__NoteWrapper-sc-2z8c2g-0 hYtmWh"
                            color="white"
                          >
                            <Notestyle__NoteTitle>
                              <span
                                className="Notestyle__NoteTitle-sc-2z8c2g-1 gNJdcF"
                              >
                                Test note 2
                              </span>
                            </Notestyle__NoteTitle>
                          </div>
                        </Notestyle__NoteWrapper>
                      </span>
                    </Blueprint3.ResizeSensor>
                  </InnerReference>
                </Reference>
                <Blueprint3.Overlay
                  autoFocus={true}
                  backdropClassName="bp3-popover-backdrop"
                  backdropProps={Object {}}
                  canEscapeKeyClose={true}
                  canOutsideClickClose={true}
                  enforceFocus={true}
                  hasBackdrop={false}
                  isOpen={false}
                  lazy={true}
                  onClose={[Function]}
                  transitionDuration={300}
                  transitionName="bp3-popover"
                  usePortal={true}
                />
              </span>
            </Manager>
          </Blueprint3.Popover>
        </Note>
      </div>
    </DocContainerstyle__DocContainerWrapper>
  </DocContainer>
</ContextProviderWrapper>
`;
